src/core/tsi/ssl_session_cache.h

```diff
@@ -0,0 +1,93 @@
+/*
+ *
+ * Copyright 2018 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef GRPC_CORE_TSI_SSL_SESSION_CACHE_H
+#define GRPC_CORE_TSI_SSL_SESSION_CACHE_H
+
+#include <grpc/support/port_platform.h>
+
+#include <grpc/slice.h>
+#include <grpc/support/sync.h>
+
+extern "C" {
+#include <openssl/ssl.h>
```

Comment:
Same ssl_session.h, please keep this header in C and make it independent of ssl.h.

Basically, you can merge this file into ssl_session_cache.cc. But define C header here.

```
typedef struct tsi_ssl_session_lru_cache tsi_ssl_session_lru_cache；
typedef struct tsi_ssl_session_ptr tsi_ssl_session_ptr;

tsi_ssl_session_lru_cache tsi_ssl_session_lru_cache_create(size_t capacity);
size_t tsi_ssl_session_lru_cache_size(tsi_ssl_session_lru_cache* cache);
void tsi_ssl_session_lru_cache_put(tsi_ssl_session_lru_cache* cache, ...);
tsi_ssl_session_lru_cache tsi_ssl_session_lru_cache_get(const char* key);
void tsi_ssl_session_lru_cache_destroy(tsi_ssl_session_lru_cache* cache);
```